keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -0,0 +1,1458 @@
+neighbours due to incorrect limits it not exclusive to pod-level resources; it can also occur
+at with container-level limits.
+
+If a cluster is upgraded, the cluster administrator will experience pre-upgrade behavior until 
+`resources` field is specified at pod-level. 
+
+If the workloads are not using the new feature, downgrade won't be affected. However, if it is required to
+downgrade a cluster that has workloads using the pod-level limits functionality, following steps need to be
+performed:
+
+* Translate pod-level resource specifications to container-level specifications.
+
+* Delete existing workloads and recreate them with container-level specifications.
+
+* Drain the nodes one by one to prevent the new pods from being scheduled on them.
+
+* Downgrade the kubelet on the drained nodes to a version that doesn't support pod-level resource specification
+fields.
+
+* Downgrade the control plane components.
+
+* Uncordon the nodes, making it available for scheduling again.
+
+
+### Version Skew Strategy
+
+Pod-level resource specification is an opt-in feature. For this feature to work correctly, it must be enabled
+in all parts of the cluster (scheduler, API server, kubelet). However, if a node is running an older version
+of kubelet that doesn't support this feature, it will ignore the pod-level resouce specifications and continue
+to use the aggregated container-level values (backward compatible). Applications must be prepared for this version
+skew implementation.
+
+For users wanting to use this feature, it is user's responsibility to use the correct version of kube-scheduler,
+kube-apiserver and kubelet otherwise the feature will not work as expected.
+
+## Production Readiness Review Questionnaire
+
+
+### Feature Enablement and Rollback
+
+<!--
+This section must be completed when targeting alpha to a release.
+-->
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+<!--
+Pick one of these and delete the rest.
+
+Documentation is available on [feature gate lifecycle] and expectations, as
+well as the [existing list] of feature gates.
+
+[feature gate lifecycle]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[existing list]: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/
+-->
+
+- [X] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: PodLevelResources
+  - Components depending on the feature gate: kubelet, kube-apiserver, kube-scheduler
+  - Will enabling / disabling the feature require downtime of the control
+    plane? No. `resources` must be set/unset in `spec` at pod-level to enable/disable this feature.
+    The workloads need to restart to disable the feature.
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node? No. Only the workloads need to restart.
+
+###### Does enabling the feature change any default behavior?
```

Comment:
this needs to be mentioned: https://github.com/kubernetes/enhancements/pull/4678/files#r1645195855

Reply:
this is still an open question here, correct? If LimitRanger was configured and somebody is trying this feature, they can work around limit rangers by setting the higher pod request than they allowed by limit ranger and quota since those will not support pod level resources initially. Not a big deal for alpha, just worth mentioning

Reply:
@jpbetz this question is for breaking change. Does any default behavior change if the feature is turned on, but not used deliberately...

Reply:
I agree this is an interesting issue that should be tracked in post-GA criteria.

I don't consider it blocking for this PRR question, since the goal here is to identify if this enhancement is fully backward compatible with pre-existing system behaviors, and this is more of a question of how it integrates with other system behaviors when used (which is interesting and important, but not what PRR is concerned with for this question specifically)

Reply:
I think handling this for beta (or at least documenting) is important. If a new feature allows to work around limits set by cluster administrator via LimitRanger, it creates a risk in rolling out the feature. So cluster administrators may need to get prepared for the rollout.

I would think it is important for beta (enabled by default) PRR.